go/mcap/cmd/doctor.go

```diff
@@ -0,0 +1,178 @@
+package cmd
+
+import (
+	"os"
+	"io"
+	"bytes"
+	"fmt"
+	"encoding/binary"
+
+	"github.com/fatih/color"
+	"github.com/foxglove/mcap/go/libmcap"
+	"github.com/spf13/cobra"
+)
+
+type mcapRecordReader struct {
+	reader       io.Reader
+}
+
+type RawRecord struct {
+	OpCode libmcap.OpCode
+	Bytes []byte
+}
+
+func (this *mcapRecordReader) Warn(format string, v ...interface{}) {
+	color.Yellow(format, v...)
+}
+
+func (this *mcapRecordReader) Error(format string, v ...interface{}) {
+	color.Red(format, v...)
+}
+
+func (this *mcapRecordReader) Fatal(v ...interface{}) {
+	color.Set(color.FgRed)
+	fmt.Println(v...)
+	color.Unset()
+	os.Exit(1)
+}
+
+func (this *mcapRecordReader) Fatalf(format string, v ...interface{}) {
+	color.Red(format, v...)
+	os.Exit(1)
+}
+
+
+func (this *mcapRecordReader) ConsumeMagic() {
+	magicBytes := make([]byte, len(libmcap.Magic))
+
+	bytesRead, err := this.reader.Read(magicBytes)
```

Comment:
It seems like you're handling it here, but one of the subtleties it's important to be aware of is that calling Read on a buffer does not guarantee that the buffer is filled - it might take multiple Read calls to fill a buffer, and depending on the reader or what is backing the reader there can be numerous reasons for this. You can address this by ensuring the bytes read is expected as you're doing here, or another alternative that will do the length check for you, is this:

```
n, err := io.ReadFull(this.reader, magicBytes)
```
which will perform multiple Read calls as necessary to fill the buffer, and return an error if fewer than len(magicBytes) can be read.